mbtools/fetch_im_resources.py

```diff
@@ -0,0 +1,63 @@
+import argparse
+from lxml import html
+from pathlib import Path
+import requests
+from mbtools.utils import parse_backup_elements, find_references_containing
+
+IM_PREFIX = "https://s3.amazonaws.com/im-ims-export/"
+
+
+def collect_and_write_resources(resource_urls, output_dir):
+    for url in resource_urls:
+        img_data = requests.get(url).content
+        filename = url.rsplit('/', 1)[-1]
+        with open(f'{output_dir}/{filename}', 'wb') as f:
+            f.write(img_data)
+
+
+def fetch_im_resources(content_dir, output_dir, mode):
+    elems = []
+    if (mode == 'mbz'):
+        elems.extend(parse_backup_elements(content_dir))
+        resource_urls = []
+        for elem in elems:
+            resource_urls.extend(
+                find_references_containing(elem.etree_fragments[0], IM_PREFIX)
+            )
+        collect_and_write_resources(set(resource_urls), output_dir)
+        return resource_urls
+    elif(mode == 'html'):
+        resource_urls = []
+        for item in content_dir.iterdir():
+            with open(item, 'r') as f:
+                data = f.read()
+                tree = html.fragments_fromstring(data)[0]
+                resource_urls.extend(
+                    find_references_containing(tree, IM_PREFIX)
+                    )
+        collect_and_write_resources(set(resource_urls), output_dir)
+        return resource_urls
+
+
+def main():
+    parser = argparse.ArgumentParser(description='')
+    parser.add_argument('content_path', type=str,
+                        help='relative path to unzipped mbz or html directory')
+    parser.add_argument('output_directory', type=str,
+                        help='Path to where files will be output')
+    parser.add_argument('mode', choices=['mbz', 'html'])
+
+    args = parser.parse_args()
+
+    content_path = Path(args.content_path).resolve(strict=True)
+    output_dir = Path(args.output_directory)
+    if not output_dir.exists():
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    replacements = fetch_im_resources(content_path, output_dir, args.mode)
+
+    print(f'Processed {len(replacements)} files to {output_dir}')
```

Comment:
Minor: This output is slightly incorrect since the length of the array returned by `fetch_im_resources` is >= the number of files in `output_dir` due to potential repeats. You could perhaps do `len(set(replacemetns)` instead.

Reply:
Actually, maybe it would be informative to output both values via something like:

```python
print(f'Processed {len(replacements)} URLs resulting in {len(set(replacements))} files to {output_dir}')
```